docx/header.py

```diff
@@ -34,6 +35,13 @@ def is_linked_to_previous(self):
             return True
         return False
 
+    @lazyproperty
+    def body(self):
+        """
+        BlockItemContainer instance with contents of Header
+        """
+        return BlockItemContainer(None, self)
```

Comment:
Okay, so this one is going to need to be more sophisticated. Here's what I think is the rough logic.

* Look in the sectPr for a headerReference of the right description (type="default" I think in this case).
* If not found, return None
* If it is found, get the rId from it, that's our key to opening the part.
* call self.part.related_header_body(rId).

  This method doesn't exist yet, so it will need to be mocked in the test. A grep on 'method_mock' in the tests will find examples.

  This method basically delegates getting the right header.xml part and loading it into a HeaderBody object, to the document part (DocumentPart) object. All relationship traversal and general part finagling is the responsibility of another part (happens at the part <-> part level). docx.parts.document.DocumentPart.numbering_part is some sort of example; there are a couple other examples on that class as well.

* HeaderBody will inherit from BlockItemContainer. BlockItemContainer is only a base class, not used directly. There will at some time be a HeaderPart class as well, in docx.parts.header.py, not sure when that will come in but it will be distinct from Header and HeaderBody.
* return header_body (the return value of self.part.related_header_body(rId)).

Just as a sketch I think it might look roughly like this:

```python
headerReference = self._sectPr.headerReference_of_type(self._type)
if headerReference is None:
    return None
return self.part.related_header_body(headerReference.rId)
```

The analysis document should also be elaborated a bit to show an XML specimen of an actual header (w:hdr at root), say with a single paragraph containing 'Foobar' or whatever. The rich details are all defined by BlockItemContainer so don't need to be repeated there. Also we'll want the XML Schema bits that rule w:hdr and any other bits and pieces you notice we're likely to need. And the parts of the protocol that we're into now should be added as well if they're not there yet. It might make sense to actually assemble a second document, the first for header access (getting ahold of a header) and the second for header properties (working with one once you actually have it in hand :).

I expect by now your starting to see that the analysis document is not the idle exercise it perhaps initially appeared to be, but rather is a collection of all the things we need now to complete the implementation :) In fact, I didn't do those initially, it was only the suffering (in the form of rework) for the lack of that information that drove me to that extra first step :)